src/stores/VoiceChannelStore.ts

```diff
@@ -182,22 +191,42 @@ export default class VoiceChannelStore extends EventEmitter {
         this.activeChannel.transport.reply(ev.detail, {});
     };
 
-    private onHangup = (ev: CustomEvent<IWidgetApiRequest>) => {
+    private updateDevices = async (fn: (devices: string[]) => string[]) => {
+        const devices = this.cli.getRoom(this.roomId)
```

Comment:
We might want to add a null-guard here. It looks like `roomId` can be set to `null` (as seen on line 227)